File: Desktop/COC/CS236/Project1/Date.cpp
#include <iostream>
#include <sstream>
#include "Date.h"
#include <string>

using namespace std;

Date::Date()
{
	day = 1;
	year = 1970;
	month = "January";
}
Date::Date(int da, int ye, string mo)
{
	if (da >= 1 && da <= 31)
		day = da;
	else
		cout << "Day is not a reasonable input between 1 and 31" << endl;
	if (ye >= 1970 && ye <= 2099)
		year = ye;
	else 
		cout << "Year is not a reasonable input between 1970 and 2099" << endl;
	month = mo;
}

int Date::getDay()
{
	return day;
}
int Date::getYear()
{
	return year;
}
string Date::getMonth()
{
	return month;
}

void Date::setDay(int da)
{
	if (da >= 1 && da <= 31)
		day = da;
	else
		cout << "Day is not a reasonable input between 1 and 31" << endl;
}
void Date::setYear(int ye)
{
	if (ye >= 1970 && ye <= 2099)
		year = ye;
	else
		cout << "Year is not a reasonable input between 1970 and 2099" << endl;
}
void Date::setMonth(string mo)
{
	month = mo;
}

string Date::toString()
{
	stringstream ssDay;
	ssDay << day;
	string dayString = ssDay.str();

	stringstream ssYear;
	ssYear << year;
	string yearString = ssYear.str();

	 return (month + " " + dayString + ", " + yearString);
}

int Date::getMonthNumber()
{
	const string monthArrayFull[] = {"January","February","March","April","May","June","July","August","September","October","November","December"};
	const string monthArrayAbbrev[] = { "Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec" };
	for (int i = 0; i < 12; i++)
	{
		if (monthArrayFull[i] == month || monthArrayAbbrev[i] == month)
		{
			return i + 1;
		}
	}
	return -1;

	/*if (month == "January" || month == "Jan")
	{
		return 1;
	}
	else if (month == "February" || month == "Feb")
	{
		return 2;
	}
	else if (month == "March" || month == "Mar")
	{
		return 3;
	}
	else if (month == "April" || month == "Apr")
	{
		return 4;
	}
	else if (month == "May")
	{
		return 5;
	}
	else if (month == "June" || month == "Jun")
	{
		return 6;
	}
	else if (month == "July" || month == "Jul")
	{
		return 7;
	}
	else if (month == "August" || month == "Aug")
	{
		return 8;
	}
	else if (month == "September" || month == "Sep")
	{
		return 9;
	}
	else if (month == "October" || month == "Oct")
	{
		return 10;
	}
	else if (month == "November" || month == "Nov")
	{
		return 11;
	}
	else if (month == "December" || month == "Dec")
	{
		return 12;
	}
	else
		return -1;*/
}
void Date::printDate(int format)
{
	if (format == 0)
	{
		cout << month << " " << day << ", " << year << endl;
	}
	else if (format == 1)
	{
		cout << day << " " << month << " " << year << endl;
	}
	else if (format == 2)
	{
		cout << getMonthNumber() << "-" << day << "-" << year << endl;
	}
	else if (format == 3)
	{
		cout << getMonthNumber() << "/" << day << "/" << year % 100 << endl;
	}
}

File: Desktop/AP Computer Science/AP CS Java/1st sem/Lab14/Lab14/Date.java
public class Date
{
    private int month;
    private int day;
    private int year;
    public Date(int monthx, int dayx, int yearx)
    {
        month = monthx;
        day = dayx;
        year = yearx;
    }
    public int getDay()
    {
        return day;
    }
    public int getMonth()
    {
        return month;
    }
    public int getYear()
    {
        return year;
    }
    public int nextYear()
    {
        year++;
        return year;
    }
    public int nextDay()
    {
        day++;
        if (month == 4 || month == 6 || month == 9 || month == 11)
        {
            if (day == 31)
            {
                month++;
                day = 1;
                return day;
            }
        }
        if (month == 2)
        {
            if (day == 29)
            {
                month++;
                day = 1;
                return day;
            }
        }
        if (month == 1 || month == 3 || month == 5 || month == 7 || month == 8 || month == 10 || month == 12)
        {
            if (day == 32)
            {
                month ++;
                day = 1;
                return day;
            }
        }
        return day;
    }
    public int nextDays(int n)
    {
        for (n=n; n > 0; n--)
        {
            nextDay();
        }
        return day;
    }
    public boolean equals(Date other)
    {
        if (month==(other.month) && day==(other.day) && year==(other.year))
        {
            return true;
        }
        return false;
    }
    public String toString()
    {
        String americanDate = new String(month+"/"+day+"/"+year);
        return americanDate;
    }
}
